src/core/output-layout.cpp

```diff
@@ -1119,7 +1126,9 @@ class output_layout_t::impl
 
         if (!noop_output)
         {
-            auto handle = wlr_headless_add_output(noop_backend, 1280, 720);
+            // NOOP output should be at least as large as actual screen sizes. Otherwise, when
+            // when windows are temporarily mapped to it, they will be moved/cropped to match it.
+            auto handle = wlr_headless_add_output(noop_backend, 3840, 2160);
```

Comment:
The noop output can be configured from the config file (IIRC), so this is only the default size if nothing else is set. I would rather not make it that big by default because if the user has some underpowered GPU mayybe this buffer would be too big?